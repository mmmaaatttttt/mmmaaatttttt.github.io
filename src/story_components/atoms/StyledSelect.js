import Select from "react-select";
import "react-select/dist/react-select.css";
import styled from "styled-components";
import { rhythm } from "../../utils/typography";

const StyledSelect = styled(Select)`
  flex: 1;
`;

export default StyledSelect;
