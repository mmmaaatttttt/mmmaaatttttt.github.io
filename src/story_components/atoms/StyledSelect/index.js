import Select from "react-select";
import styled from "styled-components";
import PropTypes from "prop-types";
import COLORS from "utils/styles";

const StyledSelect = styled(Select)`
  flex: 1;
  margin: ${props => props.margin}  
`;

StyledSelect.propTypes = {
  margin: PropTypes.string.isRequired
}

StyledSelect.defaultProps = {
  styles: {
    placeholder: () => ({
      color: COLORS.BLACK
    }),
    valueContainer: provided => ({
      ...provided,
      flexWrap: "nowrap"
    })
  },
  margin: "0.75rem 0 0 0"
};

export default StyledSelect;
