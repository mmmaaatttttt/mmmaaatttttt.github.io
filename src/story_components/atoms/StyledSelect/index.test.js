import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import StyledSelect from ".";

const wrap = (props = {}) => shallow(<StyledSelect {...props} />);

it("renders successfully", () => {
  const wrapper = wrap();
  expect(toJson(wrapper)).toMatchSnapshot();
});
