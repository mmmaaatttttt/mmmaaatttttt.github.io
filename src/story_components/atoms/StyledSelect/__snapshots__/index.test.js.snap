// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  flex: 1;
",
        ],
      },
      "defaultProps": Object {
        "styles": Object {
          "placeholder": [Function],
        },
      },
      "displayName": "Styled(StateManager)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
  styles={
    Object {
      "placeholder": [Function],
    }
  }
/>
`;
